app/institutions/dashboard/-components/institutional-users-list/component.ts

```diff
@@ -262,4 +269,69 @@ export default class InstitutionalUsersList extends Component<InstitutionalUsers
     clickToggleOrcidFilter(hasOrcid: boolean) {
         this.hasOrcid = !hasOrcid;
     }
+
+    @action
+    openMessageModal(userId: string) {
+        this.selectedUserId = userId;
+        this.messageModalShown = true;
+    }
+
+    @action
+    toggleMessageModal(userId: string | null = null) {
+        this.messageModalShown = !this.messageModalShown;
+        this.selectedUserId = userId;
+        if (!this.messageModalShown) {
+            this.resetModalFields();
+        }
+    }
+
+    @action
+    resetModalFields() {
+        this.messageText = '';
+        this.cc = false;
+        this.replyTo = false;
+        this.selectedUserId = null;
+    }
+
+    @action
+    updateMessageText(event: Event) {
+        this.messageText = (event.target as HTMLTextAreaElement).value;
+    }
+
+    @action
+    toggleCc() {
+        this.cc = !this.cc;
+    }
+
+    @action
+    toggleReplyTo() {
+        this.replyTo = !this.replyTo;
+    }
```

Comment:
You also shouldn't need these two `action`s, if you use the `<Input @type='checkbox'>` suggestion mentioned before